onnxscript/function_libs/torch_lib/ops/core.py

```diff
@@ -8359,6 +8360,63 @@ def aten_unique_consecutive(
     raise NotImplementedError()
 
 
+_NOT_IMPLEMENTED_UNIQUE = re.compile(
+    r"NOT_IMPLEMENTED\s*:\s*Could\s+not\s+find\s+an\s+implementation\s+for\s+Unique"
+    )
+"""
+A pattern to detect an unsupported (not implemented) Unique operator
+"""
+
+@torch_op("aten::unique", trace_only=True)
+def aten_unique(
+    self: TensorType,
+    sorted: bool = True,
+    return_inverse: bool = False,
+    return_counts: bool = False,
+    dim: Optional[int] = None,
+) -> tuple[TensorType, TensorType, TensorType]:
+    """unique(Tensor self, bool sorted=True, bool return_inverse=False, bool return_counts=False, int? dim=None) -> (Tensor, Tensor?, Tensor?)"""
+
+    try:
+        if dim is None:
+            unique_values, inverse_indices, counts = aten_unique2(self, sorted, return_inverse, return_counts)
+        else:
+            unique_values, inverse_indices, counts = aten_unique_dim(self, dim, sorted, return_inverse, return_counts)
+    except Exception as e:
+        # try to provide a more informative error message
+        if _NOT_IMPLEMENTED_UNIQUE.search(str(e)) is not None:
+            raise NotImplementedError(
+                f"'onnxruntime' does not yet support Unique(11) operator with dtype={self.dtype}'"
+                ) from e
```

Comment:
I would remove this try-catch as the function here is symbolic

Reply:
Addressed in b528a6ae902f3d7de87920ee1482a5a888cd14bc